code/datums/mocking/client.dm

```diff
@@ -35,7 +35,9 @@
 	src.key = "[key]_[mock_client_uid]"
 	ckey = ckey(key)
 
+#ifdef UNIT_TESTS // otherwise this shit can leak into production servers which is drather bad
 	GLOB.directory[ckey] = src
```

Comment:
I thought about doing this originally but I thought that mock clients were meant to try to replicate clients whenever possible, if the directory is filled with mock clients shouldn't we check where and make sure they get deleted when they aren't needed anymore?

Reply:
There is zero guarantee that people will remember to do that and I am not willing to figure out and audit every example presently.

Besides, @optimumtact would rather we just not make them on prod at all- this is a later issue.

Reply:
unfortunate